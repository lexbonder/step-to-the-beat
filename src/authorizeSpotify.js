/* eslint-disable */
import clientId from './clientId';

const redirectUri = encodeURIComponent('http://localhost:3000/login');

export const authorize = () => {
  window.location = `https://accounts.spotify.com/authorize?client_id=${clientId}&redirect_uri=${redirectUri}&response_type=token&scope=playlist-modify-public`;
};
