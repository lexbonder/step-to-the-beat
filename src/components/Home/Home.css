.Home {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  height: 65vh;
  margin-top: 33%;
}

.Home * {
  text-align: center;
}

.title {
  font-size: 40px;
  color: #4E9AF5;
}

.small-words {
  font-size: 25px;
}

.intro {
  display: inline-block;
  border: 5px solid #404040;
  padding: 5% 7%;
  width: 90%;
  margin: auto;
}

strong,
.app-name,
.log-in {
  color: #4E9AF5;
}

.app-name {
  font-weight: bold;
}

strong {
  font-size: 20px;
  display: inline-block;
  margin: 10px;
}

button.log-in {
  font-weight: bold;
  border: 3px solid #4E9AF5;
  background-color: transparent;
  padding: 20px;
  display: inline-block;
  width: fit-content;
  margin: auto;
  border-radius: 50px;
}