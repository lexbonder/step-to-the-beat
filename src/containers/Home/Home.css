/*.Home {
  background-color: #CCC7C4;
  width: fit-content;
  border-radius: 20px;
  margin: auto;
  padding: 20px;
  text-align: center;
}

.Home a {
  display: inline-block;
  margin: 20px;
}

.intro {
  text-align: center;
  display: block;
  width: 750px;
  margin: auto;
}

.log-in,
.begin {
  color: #322F30;
  display: inline-block;
  width: fit-content;
  padding: 20px;
  margin: 10px;
  border-radius: 20px;
  cursor: pointer;
  background-color: #FDAB02;
}

.log-in {
}

.begin {
  text-decoration: none;
}

.dashboard {
  display: flex;
  justify-content: space-around;
}

.dashboard a {
  background-color: #FDAB02;
  text-align: center;
  display: inline-block;
  padding: 20px;
  margin: 20px;
  border-radius: 10px;
  text-decoration: none;
  color: #322F30;
}

a.dashNav {
  background-color: #EC2B06;
  color: #CCC7C4;
}*/