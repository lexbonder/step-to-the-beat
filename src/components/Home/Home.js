import React from 'react';
import { authorize } from '../../authorizeSpotify';
import './Home.css';

const Home = () => {
  return (
    <div className='Home'>
      <h1 className='title'>
        Step <span className='small-words'>
            to the
        </span> Beat
      </h1>
      <p className='intro'>
        <span className='app-name'>Step to the Beat
        </span> finds music that fits your personal rhythm!
        <br />
        <br />
        {`We'll help you calculate your`}
        <br/>
        <strong>
          Steps Per Minute
        </strong>
        <br/>
        and give you a playlist that runs to your beat.
      </p>
      <button className='log-in' onClick={authorize}>
        Login with Spotify to begin
      </button>
    </div>
  );
};

export default Home;