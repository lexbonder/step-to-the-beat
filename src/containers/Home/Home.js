import React, { Component } from 'react';
import { connect } from 'react-redux';
import { authorize } from '../../authorizeSpotify';
import { NavLink, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import './Home.css';

export class Home extends Component {

  view = () => {
    const { loggedIn } = this.props;
    if (loggedIn) {
      return (
        <div className='dashboard'>
          <NavLink 
            activeClassName='dashNav'
            to='/favorite-playlists'>My Favorite Playlists</NavLink>
          <NavLink 
            activeClassName='dashNav'
            to='/select-spm'>Get New Playlist</NavLink>
        </div>
      );
    } 
  }

  render() {
    return (
      <div className='Home'>
        <p className='intro'>
          {`Step to the beat figures out your personal rhythm!
          We'll help you calculate your Steps Per Minute (SPM) and
          we'll give you a personalized playlist with the same Beats 
          Per Minute!`}
        </p>
        <button className='log-in' onClick={authorize}>
          Login with Spotify to begin
        </button>
      </div>
    );
  }
}

const { bool } = PropTypes;

Home.propTypes = {
  loggedIn: bool
};

export const MSTP = ({loggedIn}) => ({loggedIn});

export default withRouter(connect(MSTP)(Home));