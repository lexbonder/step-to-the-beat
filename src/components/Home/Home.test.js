/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import Home from './Home';
import * as authorizeSpotify from '../../authorizeSpotify';

describe('Home', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Home />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should call authorize when the button is clicked', () => {
    authorizeSpotify.authorize= jest.fn()
    const wrapper = shallow(<Home />)
    wrapper.find('button').simulate('click')

    expect(authorizeSpotify.authorize).toHaveBeenCalled()
  })
})