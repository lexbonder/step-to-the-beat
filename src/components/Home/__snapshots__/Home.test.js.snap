// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="title"
>
          Step 
          <span
                    className="small-words"
          >
                    to the
          </span>
           Beat
</h1>,
        <p
          className="intro"
>
          <span
                    className="app-name"
          >
                    Step to the Beat
          </span>
           finds music that fits your personal rhythm!
          <br />
          <br />
          We'll help you calculate your
          <br />
          <strong>
                    Steps Per Minute
          </strong>
          <br />
          and give you a playlist that runs to your beat.
</p>,
        <button
          className="log-in buttons"
          onClick={[Function]}
>
          Login with Spotify to begin
</button>,
      ],
      "className": "Home",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Step ",
            <span
              className="small-words"
>
              to the
</span>,
            " Beat",
          ],
          "className": "title",
        },
        "ref": null,
        "rendered": Array [
          "Step ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "to the",
              "className": "small-words",
            },
            "ref": null,
            "rendered": "to the",
            "type": "span",
          },
          " Beat",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="app-name"
>
              Step to the Beat
</span>,
            " finds music that fits your personal rhythm!",
            <br />,
            <br />,
            "We'll help you calculate your",
            <br />,
            <strong>
              Steps Per Minute
</strong>,
            <br />,
            "and give you a playlist that runs to your beat.",
          ],
          "className": "intro",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Step to the Beat",
              "className": "app-name",
            },
            "ref": null,
            "rendered": "Step to the Beat",
            "type": "span",
          },
          " finds music that fits your personal rhythm!",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "We'll help you calculate your",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Steps Per Minute",
            },
            "ref": null,
            "rendered": "Steps Per Minute",
            "type": "strong",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "and give you a playlist that runs to your beat.",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login with Spotify to begin",
          "className": "log-in buttons",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Login with Spotify to begin",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title"
>
            Step 
            <span
                        className="small-words"
            >
                        to the
            </span>
             Beat
</h1>,
          <p
            className="intro"
>
            <span
                        className="app-name"
            >
                        Step to the Beat
            </span>
             finds music that fits your personal rhythm!
            <br />
            <br />
            We'll help you calculate your
            <br />
            <strong>
                        Steps Per Minute
            </strong>
            <br />
            and give you a playlist that runs to your beat.
</p>,
          <button
            className="log-in buttons"
            onClick={[Function]}
>
            Login with Spotify to begin
</button>,
        ],
        "className": "Home",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Step ",
              <span
                className="small-words"
>
                to the
</span>,
              " Beat",
            ],
            "className": "title",
          },
          "ref": null,
          "rendered": Array [
            "Step ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "to the",
                "className": "small-words",
              },
              "ref": null,
              "rendered": "to the",
              "type": "span",
            },
            " Beat",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="app-name"
>
                Step to the Beat
</span>,
              " finds music that fits your personal rhythm!",
              <br />,
              <br />,
              "We'll help you calculate your",
              <br />,
              <strong>
                Steps Per Minute
</strong>,
              <br />,
              "and give you a playlist that runs to your beat.",
            ],
            "className": "intro",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Step to the Beat",
                "className": "app-name",
              },
              "ref": null,
              "rendered": "Step to the Beat",
              "type": "span",
            },
            " finds music that fits your personal rhythm!",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "We'll help you calculate your",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Steps Per Minute",
              },
              "ref": null,
              "rendered": "Steps Per Minute",
              "type": "strong",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "and give you a playlist that runs to your beat.",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login with Spotify to begin",
            "className": "log-in buttons",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Login with Spotify to begin",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
